File: problems/algorithms/object-order-item/js/response/products-qtd.js
function gtZero(products) {
  return products.filter(product => product.qtd > 0);
}

function subTotal(products) {
  const genSubTotal = p => ({ id: p.id, subtotal: p.qtd * p.preco });
  return products.map(genSubTotal);
}

function total(products) {
  return products.reduce((total, p) => total + p.qtd * p.preco, 0);
}

export { gtZero, subTotal, total };

File: problems/algorithms/object-products/ruby/response/src/products_qtd.rb
class Products
  def Products.gt_zero(products)
    products.select do |hash|
      hash[:qtd] > 0 
    end    
  end
  def Products.sub_total(products)
    products.map { |p| {id: p[:id], subtotal: p[:qtd] * p[:preco]}}
  end
  def Products.total(products)
    products.reduce(0) { |total, p| total + p[:qtd] * p[:preco]}
  end
end